### DISCUSSION TITLE
Unable to display image

### DISCUSSION THREAD

**Discussion by kokaneka** [vercel/next.js#13892] [Help]
I have the next-images as well as next-optimized-images installed. Here is what my next.config.js looks like:
```
const withImages = require('next-images');
const optimizedImages = require('next-optimized-images');
const noPrefix = process.env.PREFIX === 'NO_PREFIX';

module.exports = withImages({
    webpack(config, options) {
      config.module.rules.push({
        test: /\.(png|jpg|gif|svg|eot|ttf|woff|woff2)$/,
        use: {
          loader: 'url-loader',
          options: {
            limit: 100000,
            target: 'serverless'
          }
        }
      });

      return config;
    },
    assetPrefix: noPrefix ? '' : '/portfolio/',
  }
)
```

But still, when I try to display any image like this:
```
<img src={require('/static/images/app-mockup.png')} alt="" tw="w-64"/>
```
I get the following error:
Can't resolve '/static/images/app-mockup.png' in '/Users/kapeel.kokane/MyFiles/Code/next-blog/src/components/cta'

My other question is:
Why should the file path go through a "require"? why can't we directly use something like:
```
<img src={'/static/images/app-mockup.png'} alt="" tw="w-64"/>
```
without the require? Anyways I've tried that and it doesn't work.
As mentioned in the docs, I have a static folder inside my /public folder which also contains the images subfolder. 

I have a jsonconfig inside which I am configuring the baseUrl like so:
```
{
  "compilerOptions": {
    "baseUrl": "src"
  },
  "include": [
    "src",
  ]
}
```

should I also add public folder to this config in order for this to work?
Also, I have a custom babelrc for using twin.macro:
```
{
  "presets": [
    "next/babel",
    "@emotion/babel-preset-css-prop"
  ],
  "plugins": [
    "babel-plugin-macros"
  ]
}
```
Can this have any impact on what I am trying to do?
Please help.

**Comment by kokaneka** [comment]
Also, is there a way to import images, like svgs in this way as with create-react-app:
```
import { ReactComponent as SvgDecoratorBlob2 } from "/images/dot-pattern.svg";
```

Or is it not possible?

**Comment by todortotev** [comment]
can you provide a repo?

**Reply by kokaneka** [reply]
Sure. 

The 'working_images' branch in this repo:
https://github.com/kokanek/portfolio/tree/working_images

**Reply by todortotev** [reply]
At a project I am using, I have 100% identical to your settings and the only difference is that I don't import the images as you do, but instead I use this syntax ``` url("images/${props => props.identifier}.jpg");```

**Reply by kokaneka** [reply]
Did not get that part. Do you have this <code>url("images/${props => props.identifier}.jpg");</code> in your config or do you do this for every import?
Please don't mind, I'm quite new to nextjs

**Reply by todortotev** [reply]
I do that for every import, and I do not import them the way you do at the top of the module, rather than using the syntax for each ```<img src="url()"/>```... etc. I don't mind at all, hope we can get it solved.

**Comment by rafaelalmeidatk** [comment]
```js
<img src={require('/static/images/app-mockup.png')} alt="" tw="w-64"/>
```
This code is requiring a file from the `/static/images` folder in your system. Notice that since you are starting the path with `/` it is starting from the root which is not what you may be expecting. This path is relative to the file so you may want something like `require('../static/images/...')`, notice the `..` to go a directory up, it is exactly like how you import components. But I don't recommend including these images in the public folder, I will explain this later.

```js
Why should the file path go through a "require"? why can't we directly use something like:
<img src={'/static/images/app-mockup.png'} alt="" tw="w-64"/>
without the require? Anyways I've tried that and it doesn't work.
```

You can do it, this is the standard way of loading images. If this is not working you are not using the correct path.

<hr />

Diving a little more: importing images using `require`/`import` is only possible by using webpack loaders that change how Node module resolution works. You installed the `next-optimized-images` library that does exactly this. And also installed the `next-images` that does the same thing and you are doing it yourself all in the same project. Please check the readme of `next-optimized-images` to [setup it properly](https://github.com/cyrilwanner/next-optimized-images#installation).

When you `require`/`import` an image, these loaders will basically copy the file to the public folder with a random hash and return the path for this file. This is the reason why I recommend keeping these images outside the public folder, because the loaders will add them automatically when they are imported.

**Reply by kokaneka** [reply]
Hey, thanks a lot for that explanation. It makes things a lot more clear now from an understanding standpoint. The objective behind giving image paths in this manner 
```
/static/images/app-mockup.png
```
Was that I read this article:
https://nextjs.org/docs/basic-features/static-file-serving and though that as my **static** folder is inside the **public** folder, I can use anything inside of it in this manner. 
But seems like my understanding is not correct. 
Also, if I am using next-images/next-optimized images, I guess from your explanation, it does not matter where the image exists, it should work if the path is specified correctly, right?
I checked out my next.config and there does not seem to be anything that I was doing wrong either. 
I am trying to convert a <code>create-react-app</code> based project to a next project and hence facing all these issues. 
One final doubt. Even if I get the path import working, I see that images are being used in a way where they are being treated as modules which is a problem I will need to solve next:
```
import googlePlayIconImageSrc from "/images/google-play-icon.png"; 
...
<img src={googlePlayIconImageSrc} alt=""/>
```
I'm pretty sure there is some webpack magic going on over there! Any idea how to get **that** kind of a setup with next?

**Reply by rafaelalmeidatk** [reply]
You are confusing two kind of paths: browser resource paths and Node.js module paths. The doc page you link explains that anything you insert into the public folder is available at the root of your host. This mean that if you have a `images` folder inside the `public` folder, then yeah you can do this:
```js
<img src="/images/my-image.png" alt="my image" />
```
But notice that the `src` prop is using the **browser path**, this means it will search the file in `yourapp.com/images/my-image.png`.

When you try to import your image with `require`/`import` you are using Node.js module paths, so if you try to do this:
```js
const image = require('/images/my-image.png');
```
You are trying to access an image in your system, that is inside an `images` folder placed in the root of your system files. I am sure this is not what you want. Instead, you should give it a relative path from the file you are writing to the image file:
```js
const image = require('../images/my-image.png');
```
This will require an image that is one directory up, inside the `images` folder.

> Also, if I am using next-images/next-optimized images, I guess from your explanation, it does not matter where the image exists, it should work if the path is specified correctly, right?

Yes, these loaders will automatically move the images to the `public` folder, so you can store them anywhere in your project.

**Reply by kokaneka** [reply]
Thanks a lot for the help. Will give it a try.

**Reply by staticish** [reply]
did it help?


> Thanks a lot for the help. Will give it a try.

**Reply by barhouum7** [reply]
Vercel accepts .svg images, and the path simply should be starting from the base Url (`/`), static files, like images, should be under a folder called public in your root directory, and files inside public can then be referenced in your code starting from the base URL (`/`) 

e.g `<img src="/imgs/logo.svg" alt="My Logo" />`

This helped me, hope it helps you as well

**Reply by WinsaDaniel** [reply]
quit confused,  did you change the image file type locally?
